core/src/main/java/org/apache/brooklyn/util/core/xstream/XmlUtil.java

```diff
@@ -65,7 +66,7 @@ public static String xpath(String xml, String xpath) {
     public static Object xpath(String xml, String xpath, QName returnType) {
         try {
             DocumentBuilder builder = SharedDocumentBuilder.get();
-            Document doc = builder.parse(new ByteArrayInputStream(xml.getBytes()));
+            Document doc = builder.parse(new ByteArrayInputStream(xml.getBytes(StandardCharsets.UTF_8)));
```

Comment:
Better use `new InputSource( new StringReader( xml ) );`. Otherwise it's not clear how `builder.parse(InputStream)` will decode the input. Probably will use the standard system encoding again (since we don't have prologues on the files).